sourced/ml/cmd_entries/repos2bow.py

```diff
@@ -0,0 +1,45 @@
+import logging
+from uuid import uuid4
+
+from scipy import sparse
+
+from sourced.ml.extractors import __extractors__
+from sourced.ml.models import BOW, OrderedDocumentFrequencies, DocumentFrequencies
+from sourced.ml.transformers import Engine, UastExtractor, UastDeserializer, Uast2DocFreq, \
+    Uast2TermFreq, HeadFiles, TFIDF
+from sourced.ml.utils import create_engine
+
+
+def repos2bow_entry(args):
+    log = logging.getLogger("repos2bow")
+    engine = create_engine("repos2bow-%s" % uuid4(), **args.__dict__)
+    document_column_name = "repository_id"
+    extractors = [__extractors__[s](
+        args.min_docfreq, **__extractors__[s].get_kwargs_fromcmdline(args))
+        for s in args.feature]
+
+    uast_pipeline = Engine(engine, explain=args.explain) \
+        .link(HeadFiles()) \
+        .link(UastExtractor(languages=args.languages)) \
+        .link(UastDeserializer())
+    df_pipeline = uast_pipeline.link(Uast2DocFreq(extractors, document_column_name))
+    df = df_pipeline.execute()
+    tf_pipeline = uast_pipeline.link(Uast2TermFreq(extractors, document_column_name))
+    tf = tf_pipeline.execute()
+    tfidf = TFIDF(df, tf).execute()
+
+    tfidf = tfidf.take(tfidf.count())
```

Comment:
WAT

Reply:
calculate tfidf and get all values from rdd. what is wrong?

Reply:
Do not re-use `tfidf` for `tfidf.take`, very confusing